### PR TITLE
Add the Trace Teader to HTTP Response Headers.

### PR DESCRIPTION
**Pull request by iantalarico**
This is required to allow load balancers to make trace decisions properly.

### CODE REVIEW

File: apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/Trace/CloudTraceMiddleware.cs [hunk 1]
```diff
@@ -68,6 +68,12 @@ internal sealed class CloudTraceMiddleware
             }
             else
             {
+                // Set the trace updated trace header on the response.
+                var updatedHeaderContext = TraceHeaderContext.Create(
+                    tracer.GetCurrentTraceId(), tracer.GetCurrentSpanId() ?? 0, true);
+                httpContext.Response.Headers.Add(
+                        TraceHeaderContext.TraceHeader, updatedHeaderContext.ToString());
```
Comment (jskeet) [inline apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/Trace/CloudTraceMiddleware.cs]: Nit: outdent by 4, or just put it all on the same line.

**Comment by jskeet** [comment]
Does this leak information that users may not want leaked though? Perhaps it should be configurable? Maybe that could come in a later step - the code looks okay to me at the moment.

File: apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/Trace/CloudTraceMiddleware.cs (hunk 1 above)
Comment (iantalarico) [inline apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/Trace/CloudTraceMiddleware.cs]: Fixed

**Comment by iantalarico** [comment]
I think we always need this on but should only respond with the header if we received one.  Then we are not giving anything back we did not get (besides the decision to trace or not).

**Pull request merged**